server/services/star.js

```diff
@@ -255,7 +255,10 @@ module.exports = class StarService extends EventEmitter {
         // If both stars have warp gates and they are both owned by players...
         if (sourceStar.warpGate && destinationStar.warpGate && sourceStar.ownedByPlayerId && destinationStar.ownedByPlayerId) {
             // If both stars are owned by the player then carriers can always move at warp.
-            if (destinationStar.ownedByPlayerId.equals(player._id) && sourceStar.ownedByPlayerId.equals(player._id)) {
+            let sourceAllied = (player?.diplomacy?.allies.includes(sourceStar.ownedByPlayerId)) || sourceStar.ownedByPlayerId.equals(carrier.ownedByPlayerId)
```

Comment:
Use the `diplomacyService`, we don't want to duplicate ally check logic.

Reply:
Will you then change the rest of the code that uses warpgate?

Reply:
Because `diplomacyService` uses game, what is not as argument given to warpgate.